libnavigation-core/src/main/java/com/mapbox/navigation/core/reroute/NavigationRerouteControllerV2.kt

```diff
@@ -0,0 +1,31 @@
+package com.mapbox.navigation.core.reroute
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.route.NavigationRoute
+
+@ExperimentalPreviewMapboxNavigationAPI
+interface NavigationRerouteControllerV2 : NavigationRerouteController {
+    fun reroute(params: RerouteParameters, callback: NavigationRerouteController.RoutesCallback)
```

Comment:
Should we deprecate the old reroute method? And the interface? Because now it's not very clear how the user would implement the old reroute method. They'll still have to search for relevant request data on their own. With the deprecation it becomes clear that the old method will not be called for `NavigationRerouteControllerV2`.

Reply:
The SDK users may have `mapboxNavigation.reroute()` calls in their code. I don't know why they might want to call it, but they can, and if they switch to `NavigationRerouteControllerV2` without implementing old reroute, this case may be broken.

Reply:
`MapboxNavigation#reroute` is private, they can't invoke it.

Reply:
sorry, I meant `mapboxNavigaiton.getRerouteController()?.reroute()`

Reply:
But if they invoke it then they have it implemented. Nothing will break here.
However, why should all the users that don't invoke reroute directly implement it? If they use `NavigationRerouteControllerV2` they will probably want to receive request data as arguments. What would they do for the old reroute method? Would they just ignore the case with switching to alternatives? Or get the data themselves via `RouteProgressObserver`? But if we deprecate it with a message that says that we won't invoke it, they won't have to implement it at all. If they invoke it somewhere in their code - let them implement it. I don't see what exactly would break here.